src/test.rs

```diff
@@ -88,7 +88,7 @@ pub fn this_host_triple() -> String {
         // For windows, this host may be different to the target: we may be
         // building with i686 toolchain, but on an x86_64 host, so run the
         // actual detection logic and trust it.
-        let tp = Box::new(currentprocess::TestProcess::default());
+        let tp = Box::<currentprocess::TestProcess>::default();
```

Comment:
Please see: https://github.com/rust-lang/rustup/pull/3096#discussion_r1005435968

Reply:
Definitely can, but it is what the stable clippy is suggesting today.
During that discussion, it was still a nightly only check.
Maybe, instead of just removing the patch, I should change it to

```rust
#[allow(clippy::box_default)]
```

For this particular location.  With a link to that discussion?
Otherwise, it may come again, as clippy will keep complaining :)

Reply:
I totally agree with the consistency argument.
But, if this is what clippy is suggesting, I wonder if most new code will be written like this now.

Reply:
I would like to see a blanket allow on this for rustup.